src/api/nutrition.ts

```diff
@@ -1,54 +1,82 @@
-        NutritionAPI.localForageStore.setItem('2019-08-13', dailyIntake('2019-08-13'));
-        NutritionAPI.localForageStore.setItem('2019-08-14', dailyIntake('2019-08-14'));
-    }
-
-    private get headers(): IHeaders {
-        return {
-            'x-app-id': appID,
-            'x-app-key': appKey,
-        };
-    }
-
-    @showLoader
-    public async getSearchItems(query: string): Promise<ISearchItem[]> {
-        const config: AxiosRequestConfig = {
-            params: {
-                q: query,
-            },
-            headers: this.headers,
-        };
-        return axios.get(API_ENDPOINTS.SEARCH, config).then((result: any) => result.data.results);
-    }
-
-    @showLoader
-    public async getItem(id: string): Promise<any> {
-        const config: AxiosRequestConfig = {
-            headers: this.headers,
-        };
-        return axios.get(`${API_ENDPOINTS.ITEM}/${id}`, config);
-    }
-
-    @showLoader
-    public async getMock(day: string): Promise<IDailyIntake> {
-        return NutritionAPI.localForageStore.getItem(day);
-    }
+  private static localForageStore: LocalForage;
+
+  constructor() {
+    NutritionAPI.localForageStore = localForage.createInstance({
+      name: 'nutritionDatabase',
+    });
+    NutritionAPI.localForageStore.setItem('2019-09-09', dailyIntake('2019-09-09'));
+  }
+
+  private get headers(): IHeaders {
+    return {
+      'x-app-id': appID,
+      'x-app-key': appKey,
+    };
+  }
+
+  @showLoader
+  public async getSearchItems(query: string): Promise<FoodListItem[]> {
+    const config: AxiosRequestConfig = {
+      params: {
+        q: query,
+        offset: 40,
+        limit: 40,
+      },
+      headers: this.headers,
+    };
+    return axios
+      .get(API_ENDPOINTS.SEARCH, config)
+      .then((result: any) => result.data.results);
+  }
+
+  @showLoader
+  public async getItem(id: string): Promise<FoodInfo> {
+    const config: AxiosRequestConfig = {
+      headers: this.headers,
+    };
+    return axios.get(`${API_ENDPOINTS.ITEM}/${id}`, config)
+      .then((v: any) => {
+        return {...v.data, resource_id: id};
+      });
+  }
+
+  @showLoader
+  public async getMock(day: string): Promise<IDailyIntake | null> {
+    return NutritionAPI.localForageStore.getItem(day);
+  }
+
+  public async updateCurrentDailyIntake(currentIntake: IDailyIntake, mealName: string, food: Food)
+    : Promise<IDailyIntake> {
+      const clonedIntake: IDailyIntake = cloneDeep(currentIntake);
+      const mealIndex = clonedIntake.meals && clonedIntake.meals
+        .findIndex((mealItem: Meal) => mealItem.name.toLowerCase() === mealName.toLowerCase());
+      if (typeof mealIndex === 'number') {
+          clonedIntake.meals[mealIndex].foods.push(food);
+      }
+      return NutritionAPI.localForageStore
+        .setItem(moment(currentIntake.day).format('YYYY-MM-DD'), clonedIntake)
+        .then((v: IDailyIntake) => {
```

Comment:
This could be simplified:

`.then((v: IDailyIntake) => v`);

Reply:
Done